app/models/journeys/further_education_payments/third_party_authorisation.rb

```diff
@@ -0,0 +1,43 @@
+module Journeys
+  module FurtherEducationPayments
+    class ThirdPartyAuthorisation
+      def initialize(user:, record:)
+        @user = user
+        @claim = record
+      end
+
+      def authorised?
+        failure_reason.nil?
+      end
+
+      def failure_reason
+        return :no_service_access unless service_access?
+        return :organisation_mismatch unless organisation_matches?
+        nil
+      end
+
+      private
+
+      attr_reader :user, :claim
+
+      def service_access?
+        dfe_sign_in_user.has_role?("claim-verifier")
```

Comment:
prob should do this once in the callback and cache it in the session, we don't want to make an api to DSI to get this info on every request.